app/main.py

```diff
@@ -2,5 +2,18 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    def wrapper(*args) -> Callable:
+
+        if args in cache_storage:
+            print("Getting from cache")
+            result = cache_storage[args]["result"]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            cache_storage[args] = {"args": args, "result": result}
```

Comment:
you shouldn't create a `result` key.
Also try to use a DRY principle: https://github.com/mate-academy/py-cache-decorator/blob/master/checklist.md#dont-repeat-yourself